### PR TITLE
Github Plugin

### PR DESCRIPTION
**Pull request by amolkokane-cirus**
Fixed an issue with macros not working
Added missing javadocs
Changed copyright to 2020

### CODE REVIEW

File: src/main/java/io/cdap/plugin/github/source/batch/GithubInputFormat.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+  @Override
+  public List<InputSplit> getSplits(JobContext jobContext) throws IOException {
+    Configuration conf = jobContext.getConfiguration();
+    String configJson = conf.get(GithubFormatProvider.PROPERTY_CONFIG_JSON);
+    GithubBatchSourceConfig config = GSON.fromJson(configJson, GithubBatchSourceConfig.class);
+
+    String url = GitHubRequestFactory.generateFirstCallUrl(config);
+    HttpRequest httpRequest = GitHubRequestFactory.buildRequest(url, config.getAuthorizationToken());
+    HttpResponse response = httpRequest.execute();
```
Comment (rmstar) [inline src/main/java/io/cdap/plugin/github/source/batch/GithubInputFormat.java:53]: Please add a comment that describes what a response looks like. 
Is pagination metadata not sent if there's only 1 page?

File: src/main/java/io/cdap/plugin/github/source/common/DatasetTransformer.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+   * Transforms {@link GitHubModel} instance to {@link StructuredRecord} instance accordingly to given schema.
+   */
+  public static StructuredRecord transform(Object model, Schema schema) {
+    try {
+      StructuredRecord.Builder builder = StructuredRecord.builder(schema);
+      Class<?> clazz = model.getClass();
+      List<Schema.Field> schemaFields = schema.getFields();
+      for (Schema.Field schemaField : schemaFields) {
+        Schema.Type schemaType = schemaField.getSchema().getType();
+        Field field = getFieldByName(schemaField.getName(), clazz);
```
Comment (rmstar) [inline src/main/java/io/cdap/plugin/github/source/common/DatasetTransformer.java:42]: Can field be null?

File: src/test/java/io/cdap/plugin/github/source/common/DatasetTransformerTest.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+  public DatasetTransformerTest(Class<?> clazz) {
+    this.clazz = clazz;
+    this.model = random.nextObject(clazz);
+  }
+
+  @Parameterized.Parameters
+  public static Collection<Object[]> data() {
+    return Arrays.asList(new Object[][]{
+//      {Branch.class},
```
Comment (rmstar) [inline src/test/java/io/cdap/plugin/github/source/common/DatasetTransformerTest.java:33]: why are these commented out?

File: src/main/java/io/cdap/plugin/github/source/batch/GithubInputFormat.java [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+    return substring + pageNumber;
+  }
+
+  private String getLastLink(String paginationHeader) {
+    String[] links = paginationHeader.split(",");
+    for (String link : links) {
+      if (link.contains("last")) {
+        String[] parts = link.split(";");
+        String url = parts[0];
+        return url.substring(2, url.length() - 1);
```
Comment (rmstar) [inline src/main/java/io/cdap/plugin/github/source/batch/GithubInputFormat.java:91]: Please add a comment that describes what `getLastLink` does. What does paginationHeader look like?

File: pom.xml [hunk 5]
```diff
@@ -0,0 +1,366 @@
...
+    <assertj.version>3.11.1</assertj.version>
+    <random-beans.version>3.9.0</random-beans.version>
+    <google-http-client-gson.version>1.32.1</google-http-client-gson.version>
+  </properties>
+
+  <dependencies>
+    <dependency>
+      <groupId>io.cdap.cdap</groupId>
+      <artifactId>cdap-etl-api</artifactId>
+      <version>${cdap.version}</version>
```
Comment (rmstar) [inline pom.xml:73]: scope should be provided
https://github.com/data-integrations/google-cloud/blob/develop/pom.xml#L148

Comment (amolkokane-cirus) [reply pom.xml:73]: Done

**Review by CuriousVini — APPROVED** [review]

File: src/main/java/io/cdap/plugin/github/source/batch/GithubInputFormat.java (hunk 1 above)
Comment (shiva-cirus) [reply src/main/java/io/cdap/plugin/github/source/batch/GithubInputFormat.java:53]: https://issues.cask.co/browse/PLUGIN-382

File: src/main/java/io/cdap/plugin/github/source/common/DatasetTransformer.java (hunk 2 above)
Comment (shiva-cirus) [reply src/main/java/io/cdap/plugin/github/source/common/DatasetTransformer.java:42]: https://issues.cask.co/browse/PLUGIN-383

File: src/test/java/io/cdap/plugin/github/source/common/DatasetTransformerTest.java (hunk 3 above)
Comment (shiva-cirus) [reply src/test/java/io/cdap/plugin/github/source/common/DatasetTransformerTest.java:33]: Removed all commented code as indicated by Vinisha.

File: src/main/java/io/cdap/plugin/github/source/batch/GithubInputFormat.java (hunk 4 above)
Comment (shiva-cirus) [reply src/main/java/io/cdap/plugin/github/source/batch/GithubInputFormat.java:91]: https://issues.cask.co/browse/PLUGIN-384

**Pull request merged**